File: OOP/taller2.py
class Asiento:
    def __init__(self, color, precio, registro):
        self.color = color
        self.precio = precio
        self.registro = registro
    
    def cambiarColor(self, color):
        colores =  [ "rojo", "verde", "amarillo", "negro", "blanco" ]
        if (color in colores):
            self.color = color

class Motor:
    def __init__(self, numeroCilindros, tipo, registro):
        self.numeroCilindros = numeroCilindros
        self.tipo = tipo
        self.registro = registro
        
    def cambiarRegistro(self, registro):
        self.registro = registro
        
    def asignarTipo(self, tipo):
        if (tipo == "electrico" or tipo == "gasolina"):
            self.tipo = tipo
            
    def cantidadAsientos(self):
        return 1
        
    def verificarIntegridad(self):
        return 1
        
        
    

class Auto:
    cantidadCreados = 0
    def __init__(self):
        self.modelo
        self.precio
        self.asientos
        self.marca
        self.motor
        pass

File: OOP/Taller 2/Java/Taller 2.java
class Asiento {

  String color;
  int precio;
  int registro;

  void cambiarColor(String color) {
    switch (color) {
      case "rojo":
        this.color = "rojo";
        break;
      case "verde":
        this.color = "verde";
        break;
      case "amarillo":
        this.color = "amarillo";
        break;
      case "negro":
        this.color = "negro";
        break;
      case "blanco":
        this.color = "blanco";
        break;
      default:
        break;
    }
  }
}

class Auto {

  String modelo;
  int precio;
  Asiento[] asientos;
  String marca;
  Motor motor;
  int registro;
  public static int cantidadCreados;

  int cantidadAsientos() {
    int cont = 0;
    for (var asiento : this.asientos) {
      if (asiento instanceof Asiento) {
        cont += 1;
      }
    }
    return cont;
  }

  String verificarIntegridad() {
    String mensaje = "Auto original";

    for (Asiento asiento : this.asientos) {
      if (asiento != null) {
        if (asiento.registro != this.registro) {
          mensaje = "Las piezas no son originales";
        }
      }
    }

    if (motor.registro != this.registro) {
      mensaje = "Las piezas no son originales";
    }

    return mensaje;
  }
}

class Motor {
  int numeroCilindros;
  String tipo;
  int registro;

  void cambiarRegistro(int registro) {
    this.registro = registro;
  }

  void asignarTipo(String tipo) {
    if (tipo == "electrico") {
      this.tipo = "electrico";
    }
    if (tipo == "gasolina") {
      this.tipo = "gasolina";
    }
  }
}